src/components/alerts/WalletAlert.tsx

```diff
@@ -7,50 +11,90 @@ import {
   AlertDialogOverlay,
   Box,
   Button,
+  Flex,
   Input,
   InputGroup,
   InputRightElement,
+  Text,
   useToast,
 } from "@chakra-ui/react";
 import { useSession } from "next-auth/react";
-import { useRef, useState } from "react";
+import { ChangeEvent, useRef, useState } from "react";
 import "react-datepicker/dist/react-datepicker.css";
 
 type Props = {
   isOpen: boolean;
   onCancel: () => void;
   refetch: () => void;
+  balance: number;
+};
+export type LightningResponse = {
+  status: string;
+  tag: string;
+  commentAllowed: number;
+  callback: string;
+  metadata: string;
+  minSendable: number;
+  maxSendable: number;
+  payerData: PayerData;
+  nostrPubkey: string;
+  allowsNostr: boolean;
+};
+
+export type PayerData = {
+  name: Email;
+  email: Email;
+  pubkey: Email;
+};
+
+export type Email = {
+  mandatory: boolean;
 };
 
-const WalletAlert = ({ isOpen, onCancel, refetch }: Props) => {
+const WalletAlert = ({ isOpen, onCancel, refetch, balance }: Props) => {
   const cancelRef = useRef<HTMLButtonElement>(null);
 
   const { data: sessionData } = useSession();
   const sessionDataId = sessionData?.user?.id;
   const toast = useToast();
   const [invoiceInput, setInvoiceInput] = useState("");
+  const [amountToSend, setAmountToSend] = useState<number>();
+  const [step, setStep] = useState(1);
+  const [lightningData, setLightningData] = useState<LightningResponse>();
+  const [error, setError] = useState("");
 
   const payInvoice = usePayInvoice();
-
+  const withdrawSats = useWithdrawSats();
+  const validateAddress = useValidateAddress();
   const handleClose = () => {
     setInvoiceInput("");
     onCancel();
+    setStep(1);
+  };
+  const handleAddressValidation = (e: ChangeEvent<HTMLInputElement>) => {
+    const lnAddress = e.target.value;
+    if (lnAddress.match(/^[a-z0-9\-_.]+@[a-z0-9\-_.]+/)) {
+      setError("");
+    }
+    setInvoiceInput(e.target.value);
   };
-
   const pasteInvoice = () => {
     navigator.clipboard.readText().then((text) => {
       setInvoiceInput(text);
     });
   };
 
-  const handlePayment = async () => {
+  const handleValidation = async () => {
     if (invoiceInput === "") {
       return;
     }
-    payInvoice.mutate(
+    if (!invoiceInput.match(/^[a-z0-9\-_.]+@[a-z0-9\-_.]+/)) {
```

Comment:
having a top-level variable would be better since we're using the same regex on line 76
```suggestion
    if (!invoiceInput.match(/^[a-z0-9\-_.]+@[a-z0-9\-_.]+/)) {
```